bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolExecutorTest.java

```diff
@@ -0,0 +1,45 @@
+package org.eclipse.smarthome.core.scheduler;
+
+import static org.junit.Assert.*;
+
+import java.text.ParseException;
+
+import org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolManager.ExpressionThreadPoolExecutor;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class ExpressionThreadPoolExecutorTest {
+
+    private ExpressionThreadPoolExecutor scheduler = ExpressionThreadPoolManager.getExpressionScheduledPool("test");
+    protected boolean success;
+
+    @Test
+    @Ignore
+    public void testScheduleAndRemove() throws ParseException, InterruptedException {
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                success = true;
+            }
+        };
+
+        success = false;
+        scheduler.schedule(runnable, new CronExpression("0/1 * * * * ?"));
+        Thread.sleep(1500);
+        assertTrue(success);
+
+        boolean removed = scheduler.remove(runnable);
+
+        // TODO: This currently fails - why?
+        assertTrue(removed);
+
+        success = false;
+        Thread.sleep(1500);
+        assertFalse(success);
+
+        // TODO: What is this task that remains in the queue here? Shouldn't the queue be empty?
```

Comment:
Ouch. @maggu2810 and I changed the CronExpression in https://github.com/eclipse/smarthome/pull/2270. More efficient ways possibly, but it would break a bit the common algorithm the various Expression classes use to calculate the next execution time. I need to verify it in the debugger when I have a bit more time at hand.